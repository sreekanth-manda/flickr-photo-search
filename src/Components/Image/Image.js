import React from 'react';
import shortid from 'shortid';
import getImageUrl from '../../Utils/Utils';
import flickerService from '../../Services/FlickrService';
import Link from '../Link/Link';
import './Image.scss';

class Image extends React.Component {
  constructor(props) {
    super(props);
    this.flickrGetPhotoInfo = this.flickrGetPhotoInfo.bind(this);
    this.state = {
      imageMetaData: {}
    };
  }

  flickrGetPhotoInfo() {
    return flickerService.getPhotoInfo(this.props.image.id)
      .then(jsonResp => {
        const data = {};
        data['owner'] = jsonResp.body.photo.owner.realname;
        data['dateTaken'] = jsonResp.body.photo.dates.taken;

        const tags = jsonResp.body.photo.tags.tag.reduce((acc, item) => {
          acc.push(item.raw);
          return acc;
        }, [])

        data['tags'] = tags;
        data['url'] = getImageUrl(this.props.image);

        this.setState({ imageMetaData: data });
      })
      .catch(err => {
        console.log(err);
      })
  }

  componentDidMount() {
    this.flickrGetPhotoInfo();
  }

  componentWillUpdate(prevProps) {
    if (prevProps.image.id !== this.props.image.id) {
      this.flickrGetPhotoInfo();
    }
  }

  render() {
    return (
      <div className="image-item">
        <img className="image" src={getImageUrl(this.props.image)} alt="" />
        <div className="image-metadata">
          <span>Owner: {this.state.imageMetaData.owner}</span>
          <span>Date Taken: {this.state.imageMetaData.dateTaken}</span>
          <span>Click <Link hrefText={this.state.imageMetaData.url} linkText={'here'} /> for full Image</span>
          <div>
            <span>Tags:</span>
            {this.state.imageMetaData.tags &&
              this.state.imageMetaData.tags.map((item, key) => {
                return <span className="tag">{item}{','}</span>
              })}
          </div>
        </div>
      </div>
    );
  }
}

export default Image;
