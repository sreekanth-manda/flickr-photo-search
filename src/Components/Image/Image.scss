
.image-item {
  display: flex;
  flex-direction: column;
  margin: 25px;
  width: 80%;
  box-shadow: 10px 10px 5px -5px rgba(133,128,133,1);
  background-color: darkcyan;
  color: white;
  font-size: 14px;
  font-style: italic;

  @media screen and (min-width: 600px) {
      width: 200px;
  }
}

.image {
  width: 100%;
  height: 150px;

  @media screen and (min-width: 600px) {
      width: 200px;
  }
}

.image-metadata {
  display: flex;
  flex-direction: column;
  overflow-wrap: break-word;
  margin: 20px 5px 5px 5px;
  font-size: 14px;

  & > span, div {
    margin: 5px;
  }
}

.metadata-item {
  margin: 5px;
}

.tag {
  margin: 5px;

  & ::after {
    content: ',';
  }
}