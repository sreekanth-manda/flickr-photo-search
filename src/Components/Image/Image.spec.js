import React from 'react';
import { shallow } from 'enzyme';
import Image from './Image';

const mockFlickrGetPhotoInfo = jest.fn();

describe('Image', () => {
  const compoentProps = {
    image: {
      farm: 66,
      id: "48458406181",
      isfamily: 0,
      isfriend: 0,
      ispublic: 1,
      owner: "46922592@N00",
      secret: "23a99e54c0",
      server: "65535",
      title: "blah title 1"
    }
  }

  test('should render Image component', () => {
    const wrapper = shallow(<Image {...compoentProps} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should call flickrGetPhotoInfo', () => {
    const wrapper = shallow(<Image  {...compoentProps} />);
    wrapper.instance().flickrGetPhotoInfo = mockFlickrGetPhotoInfo;
    wrapper.setProps({ image: { id: '234234' } });
    expect(mockFlickrGetPhotoInfo.mock.calls.length).toBe(1);
  });
});