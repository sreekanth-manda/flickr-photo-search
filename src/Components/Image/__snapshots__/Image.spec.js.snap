// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render Image component 1`] = `"<div class=\\"image-item\\"><img class=\\"image\\" src=\\"https://farm66.staticflickr.com/65535/48458406181_23a99e54c0.jpg\\" alt=\\"\\"/><div class=\\"image-metadata\\"><span>Owner: </span><span>Date Taken: </span><span>Click <a class=\\"link\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">here</a> for full Image</span><div><span>Tags:</span></div></div></div>"`;
